wrappers/services.go

```diff
@@ -600,6 +601,7 @@ func generateSnapSocketFiles(app *snap.AppInfo) (*map[string][]byte, error) {
 func renderListenStream(socket *snap.SocketInfo) string {
 	snap := socket.App.Snap
 	listenStream := strings.Replace(socket.ListenStream, "$SNAP_DATA", snap.DataDir(), -1)
+	listenStream = strings.Replace(listenStream, "$XDG_RUNTIME_DIR", snap.UserXdgRuntimeDir(sys.Geteuid()), -1)
```

Comment:
I was slightly uncomfortable with this change since sys.Geteuid() is going to be whatever snapd is running as, which works for both the testsuite and typical "daemons run as root" case but would not work whenever we introduce User= and Group= in the generated systemd units. In that case, we would want to adjust sys.Geteuid() to be a lookup of the specified user/group, which is ok.

As a future reminder, can you add the following comment:
```
// TODO: when we support User/Group in the generated systemd unit,
// adjust this accordingly

Reply:
@jdstrand I was thinking exactly same thing. But I was not sure if we have anything ready for this usecase. As I assume socket options will then need to expand to SNAP_USER_COMMON and SNAP_USER_DATA. But then we need way to identify what is user service should run under. And actually forbid use of SNAP_DATA/SNAP_COMMON if not running as root. 
And indeed I used sys.Geteuid() trick to get right value in test and in actual run time
I will add comment, so make sure we keep this updated when introducing serviced for users.